src/dump.c

```diff
@@ -879,6 +884,19 @@ static void jl_serialize_value_(jl_serializer_state *s, jl_value_t *v)
     else if (jl_is_lambda_info(v)) {
         writetag(s->s, jl_lambda_info_type);
         jl_lambda_info_t *li = (jl_lambda_info_t*)v;
+        jl_serialize_value(s, (jl_value_t*)li->specTypes);
+        write_int8(s->s, li->inferred);
+        if (s->mode == MODE_MODULE || s->mode == MODE_MODULE_POSTWORK) {
+            int external = li->def && !module_in_worklist(li->def->module);
+            write_uint8(s->s, external);
+            if (external) {
+                // also flag this in the backref table as special
+                uintptr_t *bp = (uintptr_t*)ptrhash_bp(&backref_table, v);
+                assert(*bp != (uintptr_t)HT_NOTFOUND);
+                *bp |= 1; assert(((uintptr_t)HT_NOTFOUND)|1);
```

Comment:
it asserts that HT_NOTFOUND is odd. I'm pretty sure that git blame would bring to mind again why this was essential